presto-main/src/main/java/com/facebook/presto/execution/CreateTableTask.java

```diff
@@ -85,16 +85,79 @@ public String explain(CreateTable statement, List<Expression> parameters)
         checkArgument(!statement.getElements().isEmpty(), "no columns for table");
 
         QualifiedObjectName tableName = createQualifiedObjectName(session, statement, statement.getName());
+
+        LinkedHashMap<String, ColumnMetadata> columns = new LinkedHashMap<>();
+        Map<String, Object> inheritedProperties = new HashMap<>();
+        prepareColumnMetadata(statement, metadata, session, tableName, columns, inheritedProperties);
+
         Optional<TableHandle> tableHandle = metadata.getTableHandle(session, tableName);
         if (tableHandle.isPresent()) {
+            TableMetadata tableMetadata = metadata.getTableMetadata(session, tableHandle.get());
             if (!statement.isNotExists()) {
-                throw new SemanticException(TABLE_ALREADY_EXISTS, statement, "Table '%s' already exists", tableName);
+                if (columnsEquals(columns.values(), tableMetadata.getColumns())) {
```

Comment:
This was a result of a discussion where we decided to be consistent with the checks we do here and in `SemiTransactionalHiveMetastore`, and also to be consistent with the error messages we report back to the client. 

Regarding the properties, I only did the basic check of the column name/type, but you are right this is sort of arbitrary, we can add that here too.